vnext/ReactUWP/Views/TextInputViewManager.cpp

```diff
@@ -232,102 +293,159 @@ void TextInputShadowNode::createView() {
-        else if (propertyValue.isNull())
-          textBox.ClearValue(winrt::TextBox::PlaceholderForegroundProperty());
-      }
-    } else if (propertyName == "scrollEnabled") {
-      if (propertyValue.isBool() &&
-          textBox.TextWrapping() == winrt::TextWrapping::Wrap) {
-        auto scrollMode = propertyValue.asBool() ? winrt::ScrollMode::Auto
-                                                 : winrt::ScrollMode::Disabled;
-        winrt::ScrollViewer::SetVerticalScrollMode(textBox, scrollMode);
-        winrt::ScrollViewer::SetHorizontalScrollMode(textBox, scrollMode);
-      }
-    } else if (propertyName == "selection") {
-      if (propertyValue.isObject()) {
-        auto selection = json_type_traits<Selection>::parseJson(propertyValue);
-
-        if (selection.isValid())
-          textBox.Select(selection.start, selection.end - selection.start);
-      }
-    } else if (propertyName == "selectionColor") {
-      if (IsValidColorValue(propertyValue))
-        textBox.SelectionHighlightColor(SolidColorBrushFrom(propertyValue));
-      else if (propertyValue.isNull())
-        textBox.ClearValue(winrt::TextBox::SelectionHighlightColorProperty());
     } else if (propertyName == "selectTextOnFocus") {
       if (propertyValue.isBool())
         m_shouldSelectTextOnFocus = propertyValue.asBool();
-    } else if (propertyName == "spellCheck") {
-      if (propertyValue.isBool())
-        textBox.IsSpellCheckEnabled(propertyValue.asBool());
-      else if (propertyValue.isNull())
-        textBox.ClearValue(winrt::TextBox::IsSpellCheckEnabledProperty());
-    } else if (propertyName == "text") {
-      if (m_mostRecentEventCount == m_nativeEventCount) {
-        if (propertyValue.isString()) {
-          auto oldValue = textBox.Text();
-          auto newValue = asHstring(propertyValue);
-          if (oldValue != newValue) {
-            textBox.Text(newValue);
-          }
-        } else if (propertyValue.isNull())
-          textBox.ClearValue(winrt::TextBox::TextProperty());
-      }
     } else if (propertyName == "mostRecentEventCount") {
       if (propertyValue.isNumber()) {
         m_mostRecentEventCount = static_cast<uint32_t>(propertyValue.asInt());
       }
+    } else if (propertyName == "secureTextEntry") {
+      if (propertyValue.isBool()) {
+        if (propertyValue.asBool()) {
+          if (m_isTextBox) {
+            winrt::PasswordBox pwBox;
+            ReparentView(pwBox);
+            m_isTextBox = false;
+            RegisterEvents();
+            control = pwBox.as<winrt::Control>();
+          }
+        } else {
+          if (!m_isTextBox) {
+            winrt::TextBox textBox;
+            ReparentView(textBox);
+            m_isTextBox = true;
+            RegisterEvents();
+            control = textBox.as<winrt::Control>();
+          }
+        }
+      }
+    } else {
+      if (m_isTextBox) { // Applicable properties for TextBox
+        auto textBox = control.as<winrt::TextBox>();
```

Comment:
I did the cast here because Control object can change in the big loop (secureTextEntry chang). But I think I can initialize at the top and update when secureTextEntry changes.

---
In reply to: [321417273](https://github.com/microsoft/react-native-windows/pull/3075#discussion_r321417273) [](ancestors = 321417273)